sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1041,12 +1070,24 @@ class Analyzer(
         localPredicateReferences -- p.outputSet
       }
 
+      var foundNonEqualCorrelatedPred : Boolean = false
+
       // Simplify the predicates before pulling them out.
       val transformed = BooleanSimplification(sub) transformUp {
         case f @ Filter(cond, child) =>
           // Find all predicates with an outer reference.
           val (correlated, local) = splitConjunctivePredicates(cond).partition(containsOuter)
 
+          // Find any non-equality correlated predicates
```

Comment:
This is a bit more concise and will terminate as soon as we find a non-equal predicate.
```scala
foundNonEqualCorrelatedPred = foundNonEqualCorrelatedPred || correlated.exists {
  case _: EqualTo | _: EqualNullSafe => false
  case _ => true
}
```